test/e2e/mnist_rayjob_raycluster_test.go

```diff
@@ -408,8 +420,42 @@ func getRayDashboardURL(test Test, namespace, rayClusterName string) url.URL {
 	test.Eventually(Ingress(test, ingress.Namespace, ingress.Name), TestTimeoutShort).
 		Should(WithTransform(LoadBalancerIngresses, HaveLen(1)))
 
-	return url.URL{
-		Scheme: "http",
-		Host:   ingress.Spec.Rules[0].Host,
+	hostname := ingress.Spec.Rules[0].Host
+	dashboardUrl, _ := url.Parse("http://" + hostname)
+	test.T().Logf("Ray-dashboard route : %s\n", dashboardUrl.String())
+
+	return dashboardUrl.String()
+}
+
+// Create ClusterQueue
+func createClusterQueue(test Test, resourceFlavor *v1beta1.ResourceFlavor, numberOfGpus int) *v1beta1.ClusterQueue {
+	cqSpec := v1beta1.ClusterQueueSpec{
+		NamespaceSelector: &metav1.LabelSelector{},
+		ResourceGroups: []v1beta1.ResourceGroup{
+			{
+				CoveredResources: []corev1.ResourceName{corev1.ResourceName("cpu"), corev1.ResourceName("memory"), corev1.ResourceName("nvidia.com/gpu")},
+				Flavors: []v1beta1.FlavorQuotas{
+					{
+						Name: v1beta1.ResourceFlavorReference(resourceFlavor.Name),
+						Resources: []v1beta1.ResourceQuota{
+							{
+								Name:         corev1.ResourceCPU,
+								NominalQuota: resource.MustParse("8"),
+							},
+							{
+								Name:         corev1.ResourceMemory,
+								NominalQuota: resource.MustParse("12Gi"),
+							},
+							{
```

Comment:
Should it be removed if `numberOfGpus` is 0?

Reply:
IIRC specifying the resource name is mandatory even if nominalquota is 0

Reply:
https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2-latest/html/working_with_distributed_workloads/configuring-distributed-workloads_distributed-workloads#configuring-quota-management-for-distributed-workloads_distributed-workloads:~:text=Specify%20the%20resource%20name%20and%20nominalQuota%20in%20the%20flavors.resources%20section%2C%20even%20if%20the%20nominalQuota%20value%20is%200.

Reply:
The cleanest solution could be to move creation of ResourceFlavor and ClusterQueue (in the end also RayCluster) into `Test<something>Gpu/Cpu` functions. Each test function would then set properties relevant just to the test function.
That way we would get rid of conditions in test, reducing its reading complexity. Though that would mean a large rewrite.
I am ok with current solution too, just thinking out loud.

Reply:
> IIRC specifying the resource name is mandatory even if nominalquota is 0

I think it is (was?) mandatory in RHOAI due to SDK passing GPU values. Not sure if it is still needed.
Anyway in real use cases for finetuning users need GPU.

Reply:
Created a jira issue for SDK to address this - https://issues.redhat.com/browse/RHOAIENG-13618 
@astefanutti @sutaakar